src/hotspot/share/nmt/regionsTree.hpp

```diff
@@ -88,7 +88,7 @@ class RegionsTree : public VMATree {
 
   template<typename F> // F == bool(*)(CommittedMemoryRegion&)
   void visit_committed_regions(position start, size_t size, F func) {
-    size_t end = start + size;
+    size_t end = start + size + 1;
```

Comment:
It's not clear to me why `+ 1` is needed. If we have a range `[start, size+start)`, will it be missed by `visit_range_in_order`?

Reply:
Yes it is missed.
In the `visit_range_in_order` code, the variable `cmp_to` is never checked against `0` (`cmp_to == 0`).